doc/sym1.sgml

```diff
@@ -0,0 +1,133 @@
+
+<sect>Memory layout<p>
+
+The ROMs and I/O areas are defined in the configuration files, as are most of the entry points for useful subroutines in the Sym-1 monitor ROM.  cc65 generated programs compiled and linked using 4k config run in the memory range of &dollar;200 - &dollar;0FFF.  The 32k config expands this range to &dollar;7FFF.  The starting memory location and entry point for running the program is &dollar;200, so when the program is transferred to the Sym-1, it is executed by typing 'g 200'.  The system returns control back to the monitor ROM when the program terminates, providing the '.' prompt. 
+
+Special locations:
+
+<descrip>
+  <tag/Text screen/
+  Conio support is not currently available for the Sym-1.  But stdio console functions are available.
+
+  <tag/Stack/
+  The C runtime stack is located at &dollar;0FFF on 4KB Syms, or at &dollar;7FFFfor 32KB systems.  The stack always grows downwards. 
+
+  <tag/Heap/
+  The C heap is located at the end of the program and grows towards the C runtime stack.
+
+</descrip><p>
+
+<sect>Platform specific header files<p>
+
+Programs containing Sym-1 code may use the <tt/sym1.h/ header file.  See the header file for more information.
+
+<sect1>Hardware access<p>
+
+The pseudo variables declared in the <tt/sym1.inc/ include file allow access to hardware located in the address space.  See the include file for more information.
+
+<sect>Loadable drivers<p>
+
+<sect1>Graphics drivers<p>
+
+No graphics drivers are currently available for the Sym-1.
+
+<sect1>Extended memory drivers<p>
+
+No extended memory drivers are currently available for the Sym-1.
+
+<sect1>Joystick drivers<p>
+
+No joystick driver is currently available for the Sym-1.
+
+<sect1>Mouse drivers<p>
+
+No mouse drivers are currently available for the Sym-1.
+
+<sect1>RS232 device drivers<p>
+
+No communication port drivers are currently available for the Sym-1.  It has only the &quot;master console&quot; e.g. stdin and stdout.
+
+<sect>Limitations<p>
+
+<sect1>Disk I/O<p>
+
+The existing library for the Sym-1 doesn't implement C file I/O.
+
+To be more specific, this limitation means that you cannot use any of the following functions (and a few others):
+
+<itemize>
+<item>fopen
+<item>fclose
+<item>fread
+<item>fwrite
+<item>...
+</itemize>
+
+<sect>Other hints<p>
+
+<sect1>sym1.h<p>
+This header exposes Sym-specific I/O functions that are useful for reading and writing its ports and front panel.  See the <tt/sym1.h/ include file for a list of the functions available.
+
+<sect2>Limited memory applications<p>
+
+As stated earlier, there are config files for 4KB and 32KB systems.  If you have 32KB RAM, then you will probably want to use the sym1-32k configuration, but if not - if you are using the sym1-4k configuration - then you may want to use functions like getchar, putchar, gets and puts rather than printf.  Printf requires about 1KB because it needs to know how to process all the format specifiers.
```

Comment:
'getchar' and 'gets' have nothing to do with 'printf'.

Reply:
True.  I updated the comments to suggest using gets and puts rather than scanf and printf when memory space is tight.